src/app/admin/group/group.component.ts

```diff
@@ -0,0 +1,274 @@
+import { Component, OnInit, ViewChild, Inject } from "@angular/core";
+import { HttpService } from "../../shared/http.service";
+import { Group, Speciality, Faculty } from "./entity.interface";
+import { MatTableDataSource, MatTable } from "@angular/material";
+import { MatPaginator } from "@angular/material/paginator";
+import {
+  MatDialog,
+  MatDialogRef,
+  MAT_DIALOG_DATA
+} from "@angular/material/dialog";
+
+export interface DialogData {
+  data: any;
+}
+
+@Component({
+  selector: "app-group",
+  templateUrl: "./group.component.html",
+  styleUrls: ["./group.component.scss"],
+  providers: [HttpService, MatDialog]
```

Comment:
Why do we need additional instances of HttpService and MatDialog for component?

Reply:
For use in component. Was it enough to use the import directive?